Sources/AWSSDKSwift/Services/S3/S3_Shapes.swift

```diff
@@ -180,7 +179,6 @@ extension S3 {
 
         public func validate(name: String) throws {
             try self.filter?.validate(name: "\(name).filter")
-            try self.storageClassAnalysis.validate(name: "\(name).storageClassAnalysis")
```

Comment:
how does removing the bucket name pattern validation patch caused this validation to be remove?

Reply:
Becuase the only thing inside storageClassAnalysis that needs validating is a bucket name. You have to dig down about 3 or 4 classes before finding it.